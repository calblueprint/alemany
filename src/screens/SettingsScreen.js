import * as React from 'react';

import AsyncStorage from '@react-native-async-storage/async-storage';
import { func, shape } from 'prop-types';
import { View, Pressable, Text, Image } from 'react-native';

import Inset from '../components/Inset';
import ViewContainer from '../components/ViewContainer';

import BeepLogo from '/Users/kenneth/Desktop/bp/alemany/assets/images/beep.png';

import { Linking } from 'react-native';

export default function SettingsScreen({ navigation }) {
  const signOut = async () => {
    await AsyncStorage.removeItem('userToken');
    navigation.navigate('AuthLoading');
  };
  return (
    <ViewContainer>
      <Inset>
        <Pressable style={{ width: '100%' }} title="Sign out" onPress={signOut}>
          <View
            style={{
              marginTop: 10,
              padding: 20,
              borderRadius: 8,
              backgroundColor: '#eaeaef',
            }}
          >
            <Text
              style={{ fontSize: 18, fontWeight: '600', textAlign: 'center' }}
            >
              Sign out
            </Text>
          </View>
        </Pressable>
        <Text
          style={{
            marginTop: 20,
            color: '#777',
          }}
        >
          Built by Cal Blueprint.
        </Text>
        <Image source={BeepLogo} />
        <View
          style={{ position: 'relative', width: '100%', height: '100%' }}
        ></View>
      </Inset>
    </ViewContainer>
  );
}

SettingsScreen.propTypes = {
  navigation: shape({
    navigate: func,
  }),
};
