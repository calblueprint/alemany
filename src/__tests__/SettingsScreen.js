import React from 'react';

import { cleanup, render } from '@testing-library/react-native';

import SettingsScreen from '../screens/SettingsScreen';

afterEach(cleanup);

describe('Settings Screen', () => {
  it('should have an attribution', () => {
    const pattern = /built by/i;

    const { toJSON, getByText } = render(
      <SettingsScreen
        navigation={{
          navigate: () => {},
        }}
      />,
    );

    const elem = getByText(pattern);

    expect(elem).toHaveTextContent(pattern);
    expect(toJSON()).toMatchSnapshot();
  });
});
