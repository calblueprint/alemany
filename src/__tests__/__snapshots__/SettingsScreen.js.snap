// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Screen should have an attribution 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "paddingTop": 0,
      "position": "relative",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "paddingHorizontal": 16,
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "width": "100%",
        }
      }
      title="Sign out"
    >
      <View
        style={
          Object {
            "backgroundColor": "#eaeaef",
            "borderRadius": 8,
            "marginTop": 10,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
              "fontWeight": "600",
              "textAlign": "center",
            }
          }
        >
          Sign out
        </Text>
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "#777",
          "marginTop": 20,
        }
      }
    >
      Built by Blueprint.
    </Text>
  </View>
</View>
`;
